[CI] SemanticTextFieldMappingLegacyFormatIT testLegacyFormatInferenceIdUpdate failing

**Build Scans:**
- [elasticsearch-intake #40729 / part5](https://gradle-enterprise.elastic.co/s/ujdjt2ok6jg6m)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:inference:internalClusterTest" --tests "org.elasticsearch.xpack.inference.integration.SemanticTextFieldMappingLegacyFormatIT.testLegacyFormatInferenceIdUpdate" -Dtests.seed=C426596266C28713 -Dtests.locale=und -Dtests.timezone=Asia/Aqtobe -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.inference.integration.SemanticTextFieldMappingLegacyFormatIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testLegacyFormatInferenceIdUpdate'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: model_settings should be present in [sparse_field] mapping
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testLegacyFormatInferenceIdUpdate
- [main] 1 consecutive failures in step part5
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
java.lang.AssertionError: model_settings should be present in [sparse_field] mapping    
at __randomizedtesting.SeedInfo.seed([C426596266C28713:D5DD0541B009576C]:0) 
at org.junit.Assert.fail(Assert.java:89)    
at org.junit.Assert.assertTrue(Assert.java:42)  
at org.elasticsearch.xpack.inference.integration.SemanticTextLegacyFormatTestCase.assertMappingModelSettings(SemanticTextLegacyFormatTestCase.java:201) 
at org.elasticsearch.xpack.inference.integration.SemanticTextFieldMappingLegacyFormatIT.testLegacyFormatInferenceIdUpdate(SemanticTextFieldMappingLegacyFormatIT.java:777)  
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -606,9 +606,6 @@ tests:
 - class: org.elasticsearch.xpack.downsample.DataStreamLifecycleDownsampleIT
   method: testUpdateDownsampleSamplingMode
   issue: https://github.com/elastic/elasticsearch/issues/150478
-- class: org.elasticsearch.xpack.inference.integration.SemanticTextFieldMappingLegacyFormatIT
-  method: testLegacyFormatInferenceIdUpdate
-  issue: https://github.com/elastic/elasticsearch/issues/150481
 - class: org.elasticsearch.xpack.stateless.recovery.BlockRefreshUponIndexCreationIT
   method: testRefreshesAreBlockedUntilBlockIsRemoved
   issue: https://github.com/elastic/elasticsearch/issues/150492

--- a/x-pack/plugin/inference/src/internalClusterTest/java/org/elasticsearch/xpack/inference/integration/SemanticTextLegacyFormatTestCase.java
+++ b/x-pack/plugin/inference/src/internalClusterTest/java/org/elasticsearch/xpack/inference/integration/SemanticTextLegacyFormatTestCase.java
@@ -15,6 +15,7 @@ import org.elasticsearch.action.get.GetResponse;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.cluster.metadata.MappingMetadata;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.CheckedRunnable;
 import org.elasticsearch.index.mapper.InferenceMetadataFieldsMapper;
 import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
 import org.elasticsearch.inference.SimilarityMeasure;
@@ -34,6 +35,7 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
 import static org.elasticsearch.xpack.inference.Utils.storeDenseModel;
@@ -186,21 +188,29 @@ public abstract class SemanticTextLegacyFormatTestCase extends ESIntegTestCase {
      * @param shouldExist whether model_settings is expected to be present
      */
     @SuppressWarnings("unchecked")
-    protected void assertMappingModelSettings(String index, String fieldName, boolean shouldExist) {
-        GetMappingsResponse mappingsResponse = client().admin().indices().prepareGetMappings(TEST_REQUEST_TIMEOUT, index).get();
-        MappingMetadata mappingMetadata = mappingsResponse.getMappings().get(index);
-        assertThat(mappingMetadata, notNullValue());
-        Map<String, Object> mappingSource = mappingMetadata.getSourceAsMap();
-        Map<String, Object> properties = (Map<String, Object>) mappingSource.get("properties");
-        assertThat("properties should not be null", properties, notNullValue());
-        Map<String, Object> fieldMapping = (Map<String, Object>) properties.get(fieldName);
-        assertThat("field mapping for [" + fieldName + "] should not be null", fieldMapping, notNullValue());
+    protected void assertMappingModelSettings(String index, String fieldName, boolean shouldExist) throws Exception {
+        CheckedRunnable<Exception> assertion = () -> {
+            GetMappingsResponse mappingsResponse = client().admin().indices().prepareGetMappings(TEST_REQUEST_TIMEOUT, index).get();
+            MappingMetadata mappingMetadata = mappingsResponse.getMappings().get(index);
+            assertThat(mappingMetadata, notNullValue());
+            Map<String, Object> mappingSource = mappingMetadata.getSourceAsMap();
+            Map<String, Object> properties = (Map<String, Object>) mappingSource.get("properties");
+            assertThat("properties should not be null", properties, notNullValue());
+            Map<String, Object> fieldMapping = (Map<String, Object>) properties.get(fieldName);
+            assertThat("field mapping for [" + fieldName + "] should not be null", fieldMapping, notNullValue());
 
-        boolean modelSettingsPresent = fieldMapping.containsKey("model_settings");
+            boolean modelSettingsPresent = fieldMapping.containsKey("model_settings");
+            if (shouldExist) {
+                assertTrue("model_settings should be present in [" + fieldName + "] mapping", modelSettingsPresent);
+            } else {
+                assertFalse("model_settings should not be present in [" + fieldName + "] mapping before first doc", modelSettingsPresent);
+            }
+        };
+
         if (shouldExist) {
-            assertTrue("model_settings should be present in [" + fieldName + "] mapping", modelSettingsPresent);
+            assertBusy(assertion, 5, TimeUnit.SECONDS);
         } else {
-            assertFalse("model_settings should not be present in [" + fieldName + "] mapping before first doc", modelSettingsPresent);
+            assertion.run();
         }
     }
 

